google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/PublisherImplTest.java

```diff
@@ -1115,7 +1126,14 @@ public void run() {
   private Builder getTestPublisherBuilder() {
     return Publisher.newBuilder(TEST_TOPIC)
         .setExecutorProvider(FixedExecutorProvider.create(fakeExecutor))
-        .setChannelProvider(TEST_CHANNEL_PROVIDER)
+        .setChannelProvider(
+            FixedTransportChannelProvider.create(GrpcTransportChannel.create(testChannel)))
         .setCredentialsProvider(NoCredentialsProvider.create());
   }
+
+  private void shutdownTestPublisher(Publisher publisher) throws InterruptedException {
+    publisher.shutdown();
+    fakeExecutor.advanceTime(Duration.ofSeconds(10));
```

Comment:
How was 10 seconds chosen? Specifically, how long does this test need to execute on average?

Reply:
The gax layer implements a Watchdog (https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/rpc/Watchdog.java) that periodically garbage collects idle streams. The check is scheduled at a fixed rate every 10 seconds.  So, I chose 10 seconds to ensure that the watchdog check completes, otherwise we have to wait the full timeout.